### DISCUSSION TITLE
Collection of use-case

### DISCUSSION THREAD

**Discussion by greenhsu123** [Green-Software-Foundation/sci#99] [General]
Starting a thread for us to collect generalised use-cases to validate SCI. As we move through different versions of the SCI, will be useful to track how each use-case measure up with a version of SCI. Attached a couple of resources/discussions that were previously discussed. 

| Version of SCI  | Use case | Result |
| ------------- | ------------- | ------------- |
| Content Cell  | Content Cell  |Content Cell  |

[Goals/criteria for a case study to validate the SCI standard](https://github.com/Green-Software-Foundation/software_carbon_intensity/discussions/52)

In progress use-cases

- [Use-Case: Measuring the total carbon emissions for a given ML workload (training/inference) ](https://github.com/Green-Software-Foundation/software_carbon_intensity/discussions/59)
- [Use Case: Measuring the total carbon emissions for Windows](https://github.com/Green-Software-Foundation/software_carbon_intensity/discussions/54) 

Proposed use-cases
- You are the open-source maintainer of TensorFlow
- You sell a SaaS ML product that uses TensorFlow
- You are a mobile app developer
- You are a web developer.
- You are the Windows team at Microsoft
- You are the Android team at Google
reference- https://github.com/Green-Software-Foundation/software_carbon_intensity/discussions/52#discussioncomment-1238787

**Comment by atg-abhishek** [comment]
Thank you for starting this @greenhsu123 - I would also say that we can start up the appropriate [wiki](https://github.com/Green-Software-Foundation/software_carbon_intensity/wiki) page for use cases as well. It has a separate clone link that you can access from there.

**Comment by jawache** [comment]
Over time the mental rubric for me has evolved into:

A Cloud App - By far the easiest, you have access to all the data and can directly measure things.
A Web App - This is a mix of client and cloud, I think Web covers all the same ground as a mobile app or a desktop app in an OS.
An Open Source Library - You won't even know everyone who's using it or even how they are using it, getting data on real-world usage is impossible, this is the most challenging use case IMO.

**Reply by atg-abhishek** [reply]
Actually @jawache when we are talking about being consequential, in the case of OSS being the most difficult as you say, perhaps that is not actually the case. Specifically, when we talk of agency and things that are under your control, for an OSS you design and control that entire piece of software and hence have agency on all its parts. But, how many people end up using it is not something you as the developer control. So, the calculation itself isn't challenging for the OSS. How many people use it isn't consequential for the developer of the OSS. **It is consequential** for a downstream developer who chooses to use that OSS and then because they have the SCI value for the OSS, they can use that in their own application's calculation. Does that make sense?

**Reply by atg-abhishek** [reply]
I guess what I'm trying to get at is that keeping consequentialism in mind, we don't want to think about how many people are choosing to use your OSS because that is a market function of how useful and desirable your OSS is and maybe has nothing to do with the calculation of the SCI itself?

**Reply by jawache** [reply]
I'm not sure I get the point your making :) but for me, the key thing to keep in mind is that the SCI is `C` per one unit of `R`. If we stick to that thinking when we can include all types of software. If we are trying to calculate a total carbon emission, like say in the GHG protocol, then that excludes things like OSS libraries or entire languages like Python since it's impossible for them to calculate a total.

**Reply by atg-abhishek** [reply]
Ah will explain more in the next meeting when I'm there but yes, in principle I'm in agreement that the framing does allow us to talk about a diversity of types of software systems which is great!

**Comment by jawache** [comment]
Oh and Python the language, because it's a totally different paradigm but is still software and can have an SCI and covers the same ground as most other languages.

**Reply by atg-abhishek** [reply]
Would love to hear how you would place Python @jawache re: [this thread](https://github.com/Green-Software-Foundation/software_carbon_intensity/discussions/99#discussioncomment-1433919)

**Comment by navveenb** [comment]
We have created a reference on how to apply SCI for a real-world use case. I have posted the use case link for review at -   https://github.com/Green-Software-Foundation/software_carbon_intensity/discussions/120
